lib/isrlib/tests/test_isrlib.c

```diff
@@ -0,0 +1,618 @@
+
+	char src_0[10] = "0123456789";
+	char dest_0[10];
+
+	memcpy_isr(dest_0, src_0, 10);
+	UK_TEST_EXPECT_BYTES_EQ(src_0, dest_0, 10);
+
+	char src_1[14] = "Hello, World!";
+	char dest_1[14];
+	char result_1[5] = "Hello";
+
+	memcpy_isr(dest_1, src_1, 5);
+	UK_TEST_EXPECT_BYTES_EQ(result_1, dest_1, 5);
+
+	char result_2[15] = "HelloHellold!";
+
+	memcpy_isr(src_1 + 5, src_1, 5);
+	UK_TEST_EXPECT_BYTES_EQ(src_1, result_2, 14);
+
+	memcpy_isr(result_2, src_1, 0);
+	UK_TEST_EXPECT_BYTES_EQ(src_1, result_2, 14);
+
+	char src_2[1024] = {'A'};
+	char dest_2[1024];
+
+	memcpy_isr(dest_2, src_2, 1024);
+	UK_TEST_EXPECT_BYTES_EQ(src_2, dest_2, 1024);
+}
+
+/*  FIXME: There is no type checking inside strlen_isr
+ *  the following comment causes crash for accessing nullptr
+ */
+
+UK_TESTCASE(isrlib_testsuite, test_strnlen_isr)
+{
+	/* ret = strnlen_isr(NULL, 1); */
+
+	char str[] = "\0";
+	size_t len = 1;
+
+	size_t ret = strnlen_isr(str, len);
+
+	UK_TEST_EXPECT_SNUM_EQ(ret, 0);
+
+	char str1[] = "Unikraft ";
+
+	len = 10;
+
+	ret = strnlen_isr(str1, len);
+
+	UK_TEST_EXPECT_SNUM_EQ(ret, 9);
+
+	len = 5;
+	ret = strnlen_isr(str1, len);
+
+	UK_TEST_EXPECT_SNUM_EQ(ret, 5);
+}
+
+/*
+ *  FIXME: There is no type checking inside strlen_isr
+ *	the following comment causes crash for accessing nullptr
+ */
+
+UK_TESTCASE(isrlib_testsuite, test_strlen_isr)
+{
+	/* ret = strlen_isr(NULL); */
+
+	char str[] = "\0";
+	size_t ret = strlen_isr(str);
+
+	UK_TEST_EXPECT_SNUM_EQ(ret, 0);
+
+	char str1[] = "Unikraft is evolution";
+
+	ret = strlen_isr(str1);
+
+	UK_TEST_EXPECT_SNUM_EQ(ret, 21);
+
+	char str2[] = "\n\n";
+
+	ret = strlen_isr(str2);
+
+	UK_TEST_EXPECT_SNUM_EQ(ret, 2);
+}
+
+/*
+ *  FIXME: There is no type checking inside strcmp_isr
+ *  the following comment causes crash for accessing nullptr
+ */
+
+UK_TESTCASE(isrlib_testsuite, test_strcmp_isr)
+{
+	/* ret = strcmp_isr(NULL, NULL); */
+
+	int ret = strcmp_isr("unikraft", "unikraft");
+
+	UK_TEST_EXPECT_SNUM_EQ(ret, 0);
+
+	ret = strcmp_isr("unikraft", "unikrafT");
+	UK_TEST_EXPECT_SNUM_GT(ret, 0);
+
+	ret = strcmp_isr("unikrafT", "unikraft");
+	UK_TEST_EXPECT_SNUM_LT(ret, 0);
+
+	ret = strcmp_isr("unikrafT", "Unikraft");
+	UK_TEST_EXPECT_SNUM_EQ(ret, 1);
+
+	ret = strcmp_isr("", "");
+	UK_TEST_EXPECT_SNUM_EQ(ret, 0);
+}
+
+/*
+ *  FIXME: There is no type checking inside strncmp_isr
+ *	the following comment causes crash for accessing nullptr
+ */
+
+UK_TESTCASE(isrlib_testsuite, test_strncmp_isr)
+{
+	/* ret = strncmp_isr(NULL, NULL, 1); */
+
+	int ret = strncmp_isr("unikraft", "unikraft", 8);
+
+	UK_TEST_EXPECT_SNUM_EQ(ret, 0);
+
+	ret = strncmp_isr("unikraft", "unikraftfft", 8);
+	UK_TEST_EXPECT_SNUM_EQ(ret, 0);
+
+	ret = strncmp_isr("unikraft", "unikraftfft", 9);
+	UK_TEST_EXPECT_SNUM_LT(ret, 0);
+
+	ret = strncmp_isr("\0", "", 1);
+	UK_TEST_EXPECT_SNUM_EQ(ret, 0);
+}
+
+/*
+ *  FIXME: There is no type checking inside memrchr_isr
+ *  the following comment causes crash for accessing nullptr
```

Comment:
What is the following comment? I think it should be place right under this for an easy understanding.

Same for the rest.